updateHostsFile.py

```diff
@@ -1296,21 +1310,21 @@ def remove_old_hosts_file(old_file_path, backup):
         Whether or not to backup the existing hosts file.
     """
 
-    # Create if already removed, so remove won't raise an error.
-    open(old_file_path, "a").close()
+    # only go through possible backup and removal if the file exists
+    if os.path.isfile(old_file_path):
 
-    if backup:
-        backup_file_path = old_file_path + "{}".format(
-            time.strftime("%Y-%m-%d-%H-%M-%S")
-        )
+        if backup:
+            backup_file_path = old_file_path + "{}".format(
+                time.strftime("%Y-%m-%d-%H-%M-%S")
+            )
 
-        # Make a backup copy, marking the date in which the list was updated
-        shutil.copy(old_file_path, backup_file_path)
+            # Make a backup copy, marking the date in which the list was updated
+            shutil.copy(old_file_path, backup_file_path)
 
-    os.remove(old_file_path)
+        os.remove(old_file_path)
 
     # Create new empty hosts file
-    open(old_file_path, "a").close()
+    open(old_file_path, "wb").close()
```

Comment:
Why is `wb` so important?

Reply:
I could be "w" probably, I didn't write the original code, but from the comments what I think the purpose was was to make sure the file exists before it's removed, then create it.  I think the second create could be removed without issue at which point this just removes the old file it it exists.

Reply:
Actually, when I try some other parameters like '--compress' things fail, because the host file is opened directly in at least 3 different places.  I'm going to make a pass to clean that up in a different PR, so will pull this part of this one.